### PR TITLE
Migrate to null safety & add parsing from String.

### PR DESCRIPTION
**Pull request by 2shrestha22**

### CODE REVIEW

File: lib/src/properties_parser.dart [hunk 1]
```diff
@@ -168,7 +183,7 @@ class Line {
   String get keyString => String.fromCharCodes(_key).trim();
 
   /// Get the value as a String.
-  String get valueString => utf8.decode(_value).trim();
+  String get valueString => utf8.decode(_value, allowMalformed: true).trim();
```
Comment (vaind) [inline lib/src/properties_parser.dart]: Not sure about `allowMalformed`. It means we're just hiding the input error. Why do you think we should do that?

File: lib/properties.dart [hunk 2]
```diff
@@ -20,13 +20,16 @@ class Properties {
...
   /// Layout manager
-  PropertiesLayout _layout;
+  PropertiesLayout? _layout;
 
   /// An internal reference to the source file.
-  String _sourceFile;
+  String? _sourceFile;
+
+  /// An internal reference to the String value.
+  String? _stringValue;
```
Comment (vaind) [inline lib/properties.dart]: we don't need this, actually - what about removing and inlining `_initFromString()`?
```suggestion
```

**Review by vaind — COMMENTED** [review]
Thanks for doing this! Looks pretty good, just some open questions.

File: lib/src/properties_parser.dart (hunk 1 above)
Comment (2shrestha22) [reply lib/src/properties_parser.dart]: In some cases user may want non UTF-8 characters to be parsed. I think we should allow user to choose to allow not UTF-8 characters or not.

Comment (vaind) [reply lib/src/properties_parser.dart]: This library doesn't support that at the moment. It would need additional tests so I'd rather not make the change since it's not related to the point of the PR (null-safety support)

File: lib/properties.dart (hunk 2 above)
Comment (2shrestha22) [reply lib/properties.dart]: I thought later it can be used in `toFile()`. I didn't understand `toFile()` method. This method can only be used if properties are loaded from file otherwise this method will throw exception.

File: lib/src/properties_parser.dart (hunk 1 above)
Comment (2shrestha22) [reply lib/src/properties_parser.dart]: Okay I will update the PR.

**Review by vaind — APPROVED** [review]
Thanks again! LGTM
